src/storage/object/delete.js

```diff
@@ -31,32 +33,41 @@ export async function deleteObject(env, daCtx, key, isMove = false) {
   await env.DA_CONTENT.delete(`${daCtx.org}/${key}`);
   await deleteFromCollab(env, tmpCtx);
 }
+
 /**
  * Deletes one or more objects in the storage. Object is specified by the key in the daCtx or a list passed in.
  * Note: folders can not be specified in the `keys` list.
  *
  * @param {Object} env the CloudFlare environment
  * @param {DaCtx} daCtx the DA Context
- * @return {Promise<{body: null, status: number}>}
+ * @param {Object} details the details about the delete operation
+ * @return {Promise<{body: {}, status: number}>}
  */
-export default async function deleteObjects(env, daCtx) {
+export default async function deleteObjects(env, daCtx, details) {
+  if (daCtx.isFile) {
+    await deleteObject(env, daCtx, daCtx.key);
+    return { body: null, status: 204 };
+  }
+
   const keys = [];
+  if (!daCtx.isFile && !details.continuationToken) {
```

Comment:
If the details.continuationToken _is_ set it's not used later on?

Reply:
No, because a delete operation makes the subsequent `list` token invalid. The purpose of this token is just to indicate that a non-initial deletion request has been made. Only on the first request do we need to delete the `.props` for a directory.

Reply:
Got it thanks!